subsystem/cluster_v2_test.go

```diff
@@ -124,4 +125,115 @@ var _ = Describe("[V2ClusterTests]", func() {
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Wait for unbinding")
+		waitForHostStateV2(ctx, models.HostStatusUnbinding, defaultWaitForHostStateTimeout, h1)
+
+		By("Host is unbound")
+		h1 = getHostV2(infraEnvID, *h1.ID)
+		Expect(h1.ClusterID).To(BeNil())
+
+		By("Other hosts are deleted")
+		_, err = userBMClient.Installer.V2GetHost(context.Background(), &installer.V2GetHostParams{
+			InfraEnvID: boundInfraEnv,
+			HostID:     *h2.ID,
+		})
+
+		Expect(err).To(HaveOccurred())
+		_, err = userBMClient.Installer.V2GetHost(context.Background(), &installer.V2GetHostParams{
+			InfraEnvID: boundInfraEnv,
+			HostID:     *h3.ID,
+		})
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("Host unbinding pending user action on cluster delete", func() {
+		//register node to the InfraEnv and get its inventory
+		By("register h1 with InfraEnv")
+		h1 = &registerHost(infraEnvID).Host
+		generateHWPostStepReply(ctx, h1, getDefaultInventory(ips[0]), "h1")
+		waitForHostStateV2(ctx, models.HostStatusKnownUnbound, defaultWaitForHostStateTimeout, h1)
+
+		//bind the 3rd node and re-register it
+		By("bind h1 to cluster")
+		bindHost(infraEnvID, *h1.ID, clusterID)
+		waitForHostStateV2(ctx, models.HostStatusBinding, defaultWaitForHostStateTimeout, h1)
+
+		By("register h1 again and define the connectivity to the other hosts")
+		h1 = &registerHostByUUID(h1.InfraEnvID, *h1.ID).Host
+
+		generateEssentialHostSteps(ctx, h1, "h1", ips[0])
+		generateFullMeshConnectivity(ctx, ips[0], h1, h2, h3)
+		waitForHostStateV2(ctx, models.HostStatusKnown, defaultWaitForHostStateTimeout, h1)
+
+		By("cluster is ready")
+		generateEssentialPrepareForInstallationSteps(ctx, h1)
+		waitForClusterState(ctx, clusterID, models.ClusterStatusReady, defaultWaitForClusterStateTimeout,
+			IgnoreStateInfo)
+
+		By("Start installation")
+		_, err := userBMClient.Installer.V2InstallCluster(ctx, &installer.V2InstallClusterParams{ClusterID: clusterID})
+		Expect(err).NotTo(HaveOccurred())
+		generateEssentialPrepareForInstallationSteps(ctx, h1, h2, h3)
+		waitForClusterState(context.Background(), clusterID, models.ClusterStatusInstalling,
+			3*time.Minute, IgnoreStateInfo)
+
+		By("Cancel installation")
+		_, err = userBMClient.Installer.V2CancelInstallation(ctx, &installer.V2CancelInstallationParams{ClusterID: clusterID})
+		Expect(err).NotTo(HaveOccurred())
+		waitForClusterState(context.Background(), clusterID, models.ClusterStatusCancelled,
+			3*time.Minute, IgnoreStateInfo)
+
+		By("Delete Cluster")
+		_, err = userBMClient.Installer.V2DeregisterCluster(ctx, &installer.V2DeregisterClusterParams{ClusterID: clusterID})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Wait for unbinding Pending User Action")
+		waitForHostStateV2(ctx, models.HostStatusUnbindingPendingUserAction, defaultWaitForHostStateTimeout, h1)
+
+		By("Host is unbound")
+		h1 = getHostV2(infraEnvID, *h1.ID)
```

Comment:
Maybe to complete the scenario you can register the host again and check that user action is not required anymore

Reply:
sure